packages/block-library/src/cover/index.php

```diff
@@ -63,7 +63,7 @@ class="wp-block-cover__image-background"
 		$content = str_replace(
 			'</span><div',
 			'</span>' . $image . '<div',
-			$content
+			preg_replace( '/>(\s)+</m', '><', $content )
```

Comment:
Can we make the regexp only target the close span- open div markup?

Reply:
(Sorry for the late reply)
Yes, this is working:
```php		
$content = preg_replace(
    '</span><div',
    '</span>' . $image . '<div',
    preg_replace( '/<\/span>(\s+)?<div/', '</span><div', $content, 1 )
);
```

Reply:
I meant use something like `preg_replace( '/<\/span>(\s)+<div/m', '/span><div', $content )`

Reply:
> I meant use something like `preg_replace( '/<\/span>(\s)+<div/m', '/span><div', $content )` to be focused on this particular thing we're replacing not the entire block html.

I understood 😄. And I was suggesting : ` preg_replace( '/<\/span>(\s+)?<div/', '</span><div', $content, 1 )`.
The `multiline` flag is not required. Also, we should use `(\s+)?` instead of  `(\s)+` to ensure that things work also when there are no characers between `</span>` and `<div/`. And add the forth parameter to force exit after the first replacement.

Reply:
Excellent! Can you add a commit with your proposed regexp?